src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -263,6 +346,43 @@ internal Location CreateLocation(LocationVersionOne v1Location)
             return location;
         }
 
+        internal Location CreateLocation(AnnotatedCodeLocationVersionOne v1AnnotatedCodeLocation)
+        {
+            Location location = null;
+
+            if (v1AnnotatedCodeLocation != null)
+            {
+                location = new Location
+                {
+                    Annotations = v1AnnotatedCodeLocation.Annotations?.SelectMany(a => a.Locations, (a, pl) => CreateRegion(pl, a.Message)).ToList(),
```

Comment:
There is a problem here caused by the poor design of the v1 `annotation` object. You have a `location` object that refers to a file, say _file1.c_. The v1 `annotation` object contained an array of `physicalLocation` objects -- which meant that an annotation on _file1.c_  could refer to a location in _file2.c_. That makes no sense. That's why we changed the design and got rid of the `annotation` object. In v2, annotations can only refer to regions in the location's own file.

So what do you do if you encounter a v1 file that makes use of this senseless feature (as your test example does)? The best you can do is
1. Copy the whole set of v1 annotations to the property bag. You're doing that.
2. Populate the v2 `location.annotations` property with _only those annotations that refer to the file specified by_ `location.physicalLocation`.

You'll never encounter this; AFAIK no existing v1 producer uses annotations at all, still less in this senseless manner. But the code may as well be prepared for it.

Reply:
Done